core/lib/zksync_core/src/gas_tracker/mod.rs

```diff
@@ -103,8 +109,20 @@ pub(crate) fn commit_gas_count_for_l1_batch(
     let total_factory_deps_len: u32 = sorted_factory_deps
         .map(|factory_dep| factory_dep.len() as u32)
         .sum();
-    let additional_calldata_bytes = metadata.initial_writes_compressed.len() as u32
-        + metadata.repeated_writes_compressed.len() as u32
+
+    // Protocol Version 17 introduces boojum, and along with boojum an update to how storage writes
+    // are communicated/compressed.
+    let state_diff_size = match header.protocol_version.unwrap() {
+        ProtocolVersionId::Version17 => {
+            panic!("add in the size of the compressed state diffs from metadata")
```

Comment:
(I didn't fully understand the previous discussion on this line, so sorry in advance): do I get it right that this panic is going to be fixed soon? If so, would it make sense to replace `panic!` with `todo!` to make the intention more explicit?

Reply:
Also, this panic makes upgrade test fail in CI. I don't like this test to rely on the fact that server works with an incremented version, but I also don't see how it can be changed.. I think the easiest solution is to wait for https://github.com/matter-labs/zksync-era/pull/259 to be merged and remove this panic